eclass/blas.eclass

```diff
@@ -0,0 +1,241 @@
+#
+# Example usage:
+# @CODE
+# BLAS_REQ_USE="int64"
+# @CODE
+# 
+# Will result in DEPENDs like.
+# @CODE
+# DEPEND="blas_refblas? ( sci-libs/blas-reference[int64] )
+# @CODE
+
+
+
+BLAS_IMPLS=(refblas openblas gotoblas mkl)
+BLAS_SUPP_IMPLS=()
+
+_blas_provider_refblas="sci-libs/blas-reference"
+_blas_provider_openblas="sci-libs/openblas"
+_blas_provider_gotoblas="sci-libs/gotoblas2"
+_blas_provider_mkl="sci-libs/mkl"
+
+_cblas_provider_refblas="sci-libs/cblas-reference"
+_cblas_provider_openblas=""
+_cblas_provider_gotoblas=""
+_cblas_provider_mkl=""
+
+function _blas_impl_valid(){
+	local impl
+	for impl in "${BLAS_IMPLS[@]}"
+	do
+		if [ "$1" == "$impl" ]
+		then
+			return 0
+		fi
+	done
+	return 1
+}
+
+function _blas_best_impl(){
+	local impl
+	
+	for impl in "${BLAS_SUPP_IMPLS[@]}"
+	do
+		if use "$(_blas_useflag_by_impl $impl)"
+		then
+			echo $impl
+			return
+		fi
+	done
+	
+	die "No matching implementation found"
+}
+
+function _blas_useflag_by_impl(){
+	echo "${@/#/blas_}"
+}
+
+function _blas_name_by_impl(){
+	eval "echo \"\${_blas_name_$1}\""
+}
+
+function _blas_usedep(){
+	local tmp
+	tmp=("$(_blas_useflag_by_impl "${BLAS_SUPP_IMPLS[@]/%/?}" )" )
+	echo "${tmp// /,}"
+}
+
+function _blas_get_depends(){
+	local impl
+	local cblas
+	local requse
+	for impl in "${BLAS_SUPP_IMPLS[@]}"
+	do
+		if [[ $BLAS_REQ_USE ]]
+		then
+			requse="[${BLAS_REQ_USE}]"
+		else
+			requse=""
+		fi
+		if [[ $BLAS_USE_CBLAS ]]
+		then
+			eval "cblas=\$_cblas_provider_$impl"
+		else
+			cblas=""
+		fi
+		if [ -n "$cblas" ]
+		then
+			cblas="${cblas}${requse}"
+		fi
+		eval "echo \"$(_blas_useflag_by_impl $impl)? ( \${_blas_provider_$impl}${requse} ${cblas} )\""
+	done
+}
+
+function _blas_set_globals(){
+	local impl
+	
+	if [[ ${BLAS_COMPAT_ALL} ]]
+	then
+		BLAS_SUPP_IMPLS=( "${BLAS_IMPLS[@]}" )
+	else
+		if [ "${#BLAS_COMPAT[@]}" -eq 0 ]
+		then
+			die "No BLAS implementations set in BLAS_COMPAT"
+		fi
+		
+		for impl in "${BLAS_COMPAT[@]}"
+		do
+			if [ "$impl" == "*" ]
+			then
+			
+				break
+			elif _blas_impl_valid "$impl"
+			then
+				BLAS_SUPP_IMPLS+=( "$impl" )
+			else
+				die "Unknown BLAS implementation ${impl}"
+			fi
+		done
+	fi
+	IUSE="${IUSE[@]} $(_blas_useflag_by_impl "${BLAS_SUPP_IMPLS[@]}")"
+	
+	BLAS_USEDEP="$(_blas_usedep)"
+	BLAS_DEPS="$(_blas_get_depends)"
+	BLAS_REQUIRED_USE="^^ ( $(_blas_useflag_by_impl "${BLAS_SUPP_IMPLS[@]}") )"
+	REQUIRED_USE=""
+	RDEPEND=""
+	if [[ ${BLAS_CONDITIONAL_FLAG} ]]
+	then
+		for flag in "${BLAS_CONDITIONAL_FLAG[@]}"
+		do
+			REQUIRED_USE="${REQUIRED_USE} ${flag}? ( ${BLAS_REQUIRED_USE} )"
+			RDEPEND="${RDEPEND} ${flag}? ( ${BLAS_DEPS} )"
+		done
+	else
+		REQUIRED_USE="${BLAS_REQUIRED_USE}"
+		RDEPEND="${BLAS_DEPS}"
+	fi
+	DEPEND="${RDEPEND}"
+}
+
+_blas_set_globals
+unset -f _blas_set_globals
+
+
+function blas_pkg_setup(){
+	BLAS_IMPL="$(_blas_best_impl)"
+	mkdir "${T}/pkg-config/"
+	local NORMAL_PKGCONFIG="${EROOT}/usr/lib/pkgconfig"
```

Comment:
This is certainly wrong and doesn't account for multilib. Not to mention I don't think it's ever necessary.

Reply:
The ${EROOT}?
I think it is necessary for cross-emerge, which I occasionally do.